### PR TITLE
Remove all `context.TODO()`

### PR DESCRIPTION
**Pull request by glrf**
## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog

Merge after https://github.com/projectsyn/lieutenant-operator/pull/201

### CODE REVIEW

**Review by cimnine — APPROVED** [review]
LGTM, but I have requested a 2nd opinion from @ccremer

**Comment by glrf** [comment]
> LGTM, but I have requested a 2nd opinion from @ccremer

Chris is on holiday. @corvus-ch could you take a look?

File: pipeline/pipeline.go [hunk 1]
```diff
@@ -42,6 +44,13 @@ type Context struct {
 	Reconciler     reconcile.Reconciler
 }
 
+func (c Context) Context() context.Context {
+	if c.Ctx != nil {
+		return context.Background()
```
Comment (bastjan) [inline pipeline/pipeline.go:49]: I'd prefer to remove this bandaid. Contexts should always be explicitly defined in my mind and not as a side effect.

I guess it is only used in tests?

Comment (glrf) [reply pipeline/pipeline.go:49]: Good point. I was unsure whether I should add this. It's just with the way we use the `pipeline.Context` it seems very easy to forget setting the actual context. But I guess we have the same problem anyway if we forget any of the other fields..

**Review by bastjan — APPROVED** [review]
LGTM

**Review by corvus-ch — APPROVED** [review]

**Comment by corvus-ch** [comment]
Look out for #204. There I have seen some `context.TODO()`.

**Pull request merged**